x-pack/legacy/plugins/monitoring/public/lib/route_init.js

```diff
@@ -41,7 +42,8 @@ export function routeInitProvider(Private, monitoringClusters, globalState, lice
           }
 
           // check if we need to redirect because of attempt at unsupported multi-cluster monitoring
-          if (!isOnPage('home') && !cluster.isSupported) {
+          const clusterSupported = cluster.isSupported || clusters.length === 1;
```

Comment:
Why was the additional `clusters.length === 1` check needed here?

Reply:
Prior to the changes here, we always went through the `/clusters` api route, which was cause [this code](https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/monitoring/server/lib/cluster/get_clusters_from_request.js#L102) to execute, which would potentially add the `.isSupported` flag to the each cluster in the list. The changes in this PR cause that code path to _not_ be executed in some page loads, therefore this `.isSupported` flag is never there. 

I decided to change the code here (instead of adding [support for `.isSupported` here](https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/monitoring/server/lib/cluster/get_clusters_from_request.js#L93)) since a single cluster means it's supported no matter what. I can be swayed either way I guess if someone feels strongly

Reply:
Gotcha, thanks. I don't have a strong opinion so I'm good with whatever you think is best.